### PR TITLE
Rust native bootstrap

### PR DESCRIPTION
**Pull request by modmuss50**
3rd rewrite of this.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        .show_confirm();
+
+    if let Err(_) = result {
+        panic!("Failed to show dialog");
+    }
+
+    match result {
+        Ok(open) => {
+            if open {
+                if !webbrowser::open("https://fabricmc.net/wiki/player:tutorials:java:windows").is_ok() {
```
Comment (i509VCB) [inline src/main.rs]: Could probably avoid the `!foo.is_ok()` instead by doing `if let Err(_) = webbrowser::open("blah") { ... }`

File: src/main.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    if let Ok(dir) = get_minecraft_installation_dir() {
+        println!("Minecraft install dir {}", dir);
+        try_minecraft_java(&dir);
+    } else {
+        println!("Could not find minecraft install dir.");
+    }
+
+    match env::var("JAVA_HOME") {
+        Ok(val) => {
+            let full_path: String = val.to_owned() + r"bin\javaw.exe";
```
Comment (i509VCB) [inline src/main.rs]: The `to_owned()` isn't needed?

Reason being you already have an owned String and you are converting to owned form again

File: src/main.rs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    let paths: [&str; 4] = [
+        r"runtime\jre-legacy\windows-x64\jre-legacy\bin\javaw.exe",
+        r"runtime\jre-legacy\windows-x68\jre-legacy\bin\javaw.exe",
+        r"runtime\jre-x64\bin\javaw.exe",
+        r"runtime\jre-x86\bin\javaw.exe"
+    ];
+
+    for path in &paths {
+        let full_path: String = dir.to_owned() + path;
+        launch_if_valid_java_installation(&full_path)
```
Comment (i509VCB) [inline src/main.rs]: I don't see a reason to pass the path as reference as you are going to end up dropping it after you leave the for loop.

So the parameter should take a regular `String`

File: build.rs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+extern crate embed_resource;
+
+fn main() {
+    // Include our program resources for visual styling and DPI awareness on windows
+    if cfg!(windows) {
+        embed_resource::compile("platform/windows/program.rc")
```
Comment (i509VCB) [inline build.rs]: make sure you include these in the gitignore, you can move them if you wish

**Comment by modmuss50** [comment]
PR updated, now sets the attributes on the exe correctly such as the icon. All that should be left is the build/deployment stuff. I plan to just use a minimal gradle script and get them on maven. (I want to build for diffrent architectures even if we dont end up using them)

File: LICENSE [hunk 5]
```diff
@@ -178,15 +178,15 @@
    APPENDIX: How to apply the Apache License to your work.
 
       To apply the Apache License to your work, attach the following
-      boilerplate notice, with the fields enclosed by brackets "[]"
+      boilerplate notice, with the fields enclosed by brackets "{}"
```
Comment (i509VCB) [inline LICENSE]: Weird find and replace?

Comment (modmuss50) [reply LICENSE]: I think I copied an older templete, does really matter.

**Comment by modmuss50** [comment]
Should be mostly good to merge now, can always improve it later.

**Pull request merged**